File: app/src/main/java/ca/coffeeshopstudio/gaminginterfaceclient/models/Command.java
package ca.coffeeshopstudio.gaminginterfaceclient.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

/**
 * TODO: HEADER COMMENT HERE.
 */
public class Command {
    public static final int KEY_DOWN = 0;
    public static final int KEY_UP = 1;
    @SerializedName("Key")
    private String key = "a";
    @SerializedName("Modifier")
    private List<String> modifiers = new ArrayList<>();
    @SerializedName("ActivatorType")
    private int activatorType; //key down, key up, etc

    public void addModifier(String modifier) {
        modifiers.add(modifier);
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public List<String> getModifiers() {
        return modifiers;
    }

    public int getActivatorType() {
        return activatorType;
    }

    public void setActivatorType(int activatorType) {
        this.activatorType = activatorType;
    }

    public void removeAllModifiers() {
        modifiers = new ArrayList<>();
    }
}

File: gic_flutter/lib/model/screen/Command.dart
class Command {
  static const int KEY_DOWN = 0;
  static const int KEY_UP = 1;

  String key = "a";
  List<String> modifiers = new List<String>();
  int activatorType; //key down, key up, etc

  Command ({this.key, this.modifiers, this.activatorType});

  Map<String, dynamic> toJson() =>
  {
    'activatorType': activatorType,
    'key': key,
    'modifiers': modifiers
  };

  factory Command.fromJson(Map<String, dynamic> json) {
    var jsonMods = json['modifiers'];
    List<String> mods = new List<String>.from(jsonMods);
    String jsonKey = json['key'];
    int jsonActivator = json['activatorType'];

    return Command(
      key: jsonKey,
      modifiers: mods,
      activatorType: jsonActivator
    );
  }

}
